opengever/workspace/vocabularies.py

```diff
@@ -74,7 +74,12 @@ def getTerm(self, value):
         try:
             return super(WorkspaceUserVocabulary, self).getTerm(value)
         except LookupError:
-            return AllUsersSource(api.portal.get()).getTermByToken(value)
+            try:
+                # Attempt to get the term from AllUsersSource
+                return AllUsersSource(api.portal.get()).getTermByToken(value)
+            except (LookupError):
+                # If term not found reutn  "not found" as a string
+                return "not found"
```

Comment:
please, return a `SimpleTerm` and a value of `<not-found>` : `return SimpleTerm(value, value, value)`

And let's do the following check here before calling the `AllUsersSource`. This let's us return the `not found` before even calling the vocabulary. We then don't need to catch the `LookupError` of the `AllUsersSource`. So something like that: 
  

```suggestion
            if not visible_users_and_groups_filter.can_access_principal(value):
                return SimpleTerm( `<not-found>`,  `<not-found>`,  `<not-found>`)
            return AllUsersSource(api.portal.get()).getTermByToken(value)
```